### PR TITLE
rsample fix for rlang::abort as cli_abort for 506 

### LINKED ISSUE #506: Use cli errors in `R/initial_validation_split.R`
Most or all errors thrown in this package are made via `rlang::abort()`. We are transitioning to `cli::cli_abort()` to make use of the richer styling options for errors via cli (which calls rlang under the hood). This issue is part of that effort (#457).

It is often sufficient to search for and replace the rlang version with the cli version together with adding a little bit of styling.

Some examples of the styling possibilities are in the [general introduction](https://cli.r-lib.org/articles/semantic-cli.html) under the section of [Inline text formatting](https://cli.r-lib.org/articles/semantic-cli.html#inline-text-formatting). More details on [inline formatting](https://cli.r-lib.org/reference/inline-markup.html) and [pluralization](https://cli.r-lib.org/articles/pluralization.html). 

One aspect to watch out for when transitioning from `rlang::abort()` to `cli::cli_abort()` are messages pre-assembled from user input as described [here](https://rlang.r-lib.org/dev/reference/topic-condition-formatting.html#transitioning-from-abort-to-cli-abort-).

#499 can serve as an example.

### LINKED ISSUE #508: Use cli errors in `R/make_groups.R`, `R/mc.R`, `R/nest.R`
Most or all errors thrown in this package are made via `rlang::abort()`. We are transitioning to `cli::cli_abort()` to make use of the richer styling options for errors via cli (which calls rlang under the hood). This issue is part of that effort (#457).

It is often sufficient to search for and replace the rlang version with the cli version together with adding a little bit of styling.

Some examples of the styling possibilities are in the [general introduction](https://cli.r-lib.org/articles/semantic-cli.html) under the section of [Inline text formatting](https://cli.r-lib.org/articles/semantic-cli.html#inline-text-formatting). More details on [inline formatting](https://cli.r-lib.org/reference/inline-markup.html) and [pluralization](https://cli.r-lib.org/articles/pluralization.html). 

One aspect to watch out for when transitioning from `rlang::abort()` to `cli::cli_abort()` are messages pre-assembled from user input as described [here](https://rlang.r-lib.org/dev/reference/topic-condition-formatting.html#transitioning-from-abort-to-cli-abort-).

#499 can serve as an example.

### PR DESCRIPTION
**Pull request by PriKalra**
replaced all the rlang::abort to cli_abort without using cli::
closes #506 and closes #508

### CODE REVIEW

**Review by PriKalra — COMMENTED** [review]
fixes #508

File: R/initial_validation_split.R [hunk 1]
```diff
@@ -120,25 +120,25 @@ initial_validation_split <- function(data,
 
 check_prop_3 <- function(prop, call = rlang::caller_env()) {
   if (!is.numeric(prop)) {
-    rlang::abort("`prop` needs to be numeric.", call = call)
+    cli_abort("`prop` needs to be numeric.", call = call)
```
Comment (hfrick) [inline R/initial_validation_split.R:123]: Could you please add the `{.arg}` styling for `prop` in the abort calls in this function? It should look like `{.arg group}`.

File: R/initial_validation_split.R [hunk 2]
```diff
@@ -303,8 +303,8 @@ validation <- function(x, ...) {
 #' @rdname initial_validation_split
 validation.default <- function(x, ...) {
   cls <- class(x)
-  cli::cli_abort(
-    "No method for objects of class{?es}: {cls}"
+  cli_abort(
+    "No method for objects of class{?es}: {.cls {cls}} "
```
Comment (hfrick) [inline R/initial_validation_split.R:307]: Nice!

File: R/make_groups.R [hunk 3]
```diff
@@ -296,7 +296,7 @@ check_prop <- function(prop, replace) {
     ((prop <= 1 && replace) || (prop < 1 && !replace))
   acceptable_prop <- acceptable_prop && prop > 0
   if (!acceptable_prop) {
-    rlang::abort(
+    cli_abort(
       "`prop` must be a number between 0 and 1.",
```
Comment (hfrick) [inline R/make_groups.R:300]: another opportunity to style `prop` as an argument

File: R/mc.R [hunk 4]
```diff
@@ -104,7 +104,7 @@ mc_complement <- function(ind, n) {
 mc_splits <- function(data, prop = 3 / 4, times = 25,
                       strata = NULL, breaks = 4, pool = 0.1) {
   if (!is.numeric(prop) | prop >= 1 | prop <= 0) {
-    rlang::abort("`prop` must be a number on (0, 1).")
+    cli_abort("`prop` must be a number on (0, 1).")
```
Comment (hfrick) [inline R/mc.R:107]: an one more for `prop`

File: R/make_groups.R [hunk 5]
```diff
@@ -345,13 +345,13 @@ validate_group <- function(group, data, call = rlang::caller_env()) {
   }
 
   if (is.null(group) || !is.character(group) || length(group) != 1) {
-    rlang::abort(
-      "`group` should be a single character value for the column that will be used for splitting.",
+    cli_abort(
+      "{.arg {group}} should be a single character value for the column that will be used for splitting.",
```
Comment (hfrick) [inline R/make_groups.R:349]: Here we only want to say the argument `group` but not what is actually inside of `group` so this should just say `{.arg group}` here.

File: R/make_groups.R [hunk 6]
```diff
@@ -345,13 +345,13 @@ validate_group <- function(group, data, call = rlang::caller_env()) {
...
       call = call
     )
   }
   if (!any(names(data) == group)) {
-    rlang::abort("`group` should be a column in `data`.", call = call)
+    cli_abort("{.arg {group}} should be a column in {.arg {data}}.", call = call)
```
Comment (hfrick) [inline R/make_groups.R:354]: Same here for `group` and `data`. We don't want to evaluate those here, that's what the inner set of `{}` does.

File: R/nest.R [hunk 7]
```diff
@@ -80,9 +80,9 @@ nested_cv <- function(data, outside, inside) {
 
   inner_cl <- cl[["inside"]]
   if (!is_call(inner_cl)) {
-    abort(
-      "`inside` should be a expression such as `vfold()` or ",
-      "bootstraps(times = 10)` instead of an existing object.",
+    cli_abort(
+      "{.arg {inside}} should be a expression such as {.fun vfold} or //
```
Comment (hfrick) [inline R/nest.R:84]: Here we want `inside` to be styled as an argument, but not evaluated, so we only need the outer set of `{}`.

**Review by hfrick — COMMENTED** [review]
Thank you for tackling this! Could you also run the tests and update snapshots accordingly? If you'd like some help doing that, come see me!

File: R/nest.R (hunk 7 above)
Comment (PriKalra) [reply R/nest.R]: changed

File: R/make_groups.R (hunk 6 above)
Comment (PriKalra) [reply R/make_groups.R]: changed

File: R/make_groups.R (hunk 5 above)
Comment (PriKalra) [reply R/make_groups.R]: changed

File: R/mc.R (hunk 4 above)
Comment (PriKalra) [reply R/mc.R]: changed

File: R/make_groups.R (hunk 3 above)
Comment (PriKalra) [reply R/make_groups.R]: changed

**Comment by PriKalra** [comment]
I have done the tests, inluded the feedback and run a rCMD check.

**Pull request merged**